Update hooks args and return values to make them more readable and add wagmi links

Pabl0cks opened the pull request:
This is a proposal to add more readability to the args and return values from `useScaffoldContractRead`, `useScaffoldContractWrite` and `useScaffoldEventHistory`  hooks. 

Also adding wagmi links to `useScaffoldContractRead`, `useScaffoldContractWrite` and `useScaffoldEventSubscriber` so our users can check configuration and types.

If you feel `h2` "Parameters" is not needed in some (or all) of them, we can delete it. Also was not very sure on how to add the wagmi reference for `useScaffoldEventSubscriber`, so just added a small sentence mentioning it.

Closes #39

technophile-04 commented on docs/hooks/useScaffoldEventHistory.md:
```diff
@@ -29,4 +29,15 @@ const {
-This example retrieves the historical event logs for the `GreetingChange` event of the `YourContract` smart contract, starting from block number 31231 and filtering events where the premium parameter is true. The data property of the returned object contains an array of event objects, each containing the event parameters and (optionally) the block, transaction, and receipt data. The `isLoading` p ...
+This example retrieves the historical event logs for the `GreetingChange` event of the `YourContract` smart contract, starting from block number 31231 and filtering events where the premium parameter is true.
+
+## Parameters
+
+All the parameters accepted by `useScaffoldEventHistory` hooks are explained in above code block example with comments.
```
Lol don't know if this is too informal, I added it at https://github.com/scaffold-eth/se2-docs/pull/40/commits/183e59b5351d30a1a1004193cd9781649a30b30f

Don't know if we should remove comments from code example and explain all the parameters here OR we could completely remove "Parameters" section for these hooks since we already described them at code block example

Or this looks fine?

Pabl0cks replied:
Thanks for the update and feedback @technophile-04 🙌

> Don't know if we should remove comments from code example and explain all the parameters here OR we could completely remove "Parameters" section for these hooks since we already described them at code block example

Reading all hooks again, I would vote to either getting rid of "Parameters" section, or removing the code comments and do a formal explanation of each parameter in the "Parameters" section. I think any of those could be a better implementation than the current "intermediate" one. 

Maybe simpler one can be just remove "Parameters", but we can wait for other opinions 🙌

carletex replied:
I'd vote for the latter! Same thing we are doing with component props:

https://docs.scaffoldeth.io/components/Address

Pabl0cks replied:
> I'd vote for the latter! Same thing we are doing with component props:

Adding it! I think it will look much better in readability terms and maintains consistency 🙌

Pabl0cks commented:
I've just pushed the Parameter tables for `useScaffoldContractRead`, `useScaffoldContractWrite`, `useScaffoldEventHistory` and `useScaffoldEventSubscriber`.

Haven't modified `useDeployedContractInfo` and `useScaffoldContract` because they are too simple, so I don't know if they would look well.

In this case I haven't add a "Default Value" column, adding the info in Description field for those parameters with default value. If you prefer to have it like in Components, let me know and I'll add it 🙌.

technophile-04 commented on docs/hooks/useScaffoldEventHistory.md:
```diff
@@ -18,15 +18,32 @@ const {
+- `data` property of the returned object contains an array of event objects, each containing the event parameters and (optionally) the block, transaction, and receipt data.
+- `isLoading` property indicates whether the event logs are currently being fetched.
+- `error` property contains any error that occurred during the fetching process (if applicable).
+- Since `watch` is set to true, the event logs will be refetched every [`pollingInterval`](/deploying/deploy-nextjs-app#--pollinginterval) (set at `scaffold.config.ts`).
```
I think we can remove this from here since we already have a `watch` desc in "Parameters" section.(we need to update `pollingInterval` link their)

Pabl0cks replied:
Removed 🙌

technophile-04 commented:
Tysm @Pabl0cks !!

> Haven't modified useDeployedContractInfo and useScaffoldContract because they are too simple, so I don't know if they would look well.

Umm just to be consistent added them at https://github.com/scaffold-eth/se2-docs/pull/40/commits/048b739c228abb819b5f369123b09f434ba1f721 😅 also added some references to wagmi/viem docs 


--- 

I think it almost looking perfect added a couple of comments and a small NITPICK would be : 

changing "Parameters" h2 to "Configurations" since the hooks actually accept one object as parameter similar to what wagmi docs does and makes sense to me 🙌. For `useDeployedContractInfo` I think its fine being "Parameters" their

Pabl0cks commented on docs/hooks/useScaffoldEventSubscriber.md:
```diff
@@ -24,6 +24,16 @@ useScaffoldEventSubscriber({
 
 This example subscribes to the `GreetingChange` event emitted by the `YourContract` smart contract and logs the parameters from the event to the console when it's emitted. The `listener` function accepts an array of `logs` that occurred during the [`pollingInterval`](/deploying/deploy-nextjs-app#--pollinginterval) (set at `scaffold.config.ts`) and each array items contains an `args` property whic ...
```
`Listener` function explanation moved to the Configuration table 👌🙌

Pabl0cks commented:
> changing "Parameters" h2 to "Configurations" since the hooks actually accept one object as parameter similar to what wagmi docs does and makes sense to me 🙌. For useDeployedContractInfo I think its fine being "Parameters" their.
> 
> And also maybe mentioning about (optional) similar to what we are doing with https://docs.scaffoldeth.io/components

- Changed h2 for the table to "Configuration". Except for `useDeployedContractInfo` that keeps "Parameters".
- Added (optional) tag to Parameter name column for those parameters that are optional.
- Also tweaked a bit some descriptions, mostly some "The.." that were too repetitive.

TYSM @technophile-04 for your help on this PR ♥

carletex commented on docs/hooks/useScaffoldContractRead.md:
Not from this PR, but it bugs me a bit that the example is not a valid one (because of the args)

We could just remove it (since it's detailed in the config) or just use a different example (get `userGreetingCounter` with a given address`

Pabl0cks replied:
Sure! 🙌 Just pushed the updated example with the working one at => 6903c496fd98b8662131e471cc4c2c896a876103

carletex commented:
This is looking great!

Left a tiny thing, but this looks good for a merge. We can iterate later :)

Thanks @Pabl0cks and @technophile-04

Pabl0cks commented:
> Left a tiny thing, but this looks good for a merge. We can iterate later :)

Cool! Thanks for the review! I'll merge it then.

If you see something wrong in the updated `useScaffoldContractRead` example or Shiv detects something to be changed, I'll create a new PR for the quickfixes.


TYSM! 🙌
